src/app/dialogs/rewards-history-dialog/rewards-history-dialog.component.ts

```diff
@@ -0,0 +1,60 @@
+import {AfterViewInit, ChangeDetectorRef, Component, Input} from '@angular/core';
+import {HttpService} from '../../services/http.service';
+import {ViewTypeService} from '../../services/view-type.service';
+import {NGXLogger} from 'ngx-logger';
+import {RewardDto} from "../../models/reward-dto";
+import {StaticValues} from '../../static/static-values';
+
+@Component({
+    selector: 'app-rewards-history-dialog',
+    templateUrl: './rewards-history-dialog.component.html',
+    styleUrls: ['./rewards-history-dialog.component.css']
+})
+
+export class RewardsHistoryDialogComponent implements AfterViewInit {
+    @Input() data;
+    rewards: Array<RewardDto> = [];
+    ready = false;
+    disabled = false;
+    vaultFilter = '';
+    filters = StaticValues.currentVaults;
+
+    constructor(private httpService: HttpService,
+                public vt: ViewTypeService,
+                private cdRef: ChangeDetectorRef,
+                private log: NGXLogger) {
+    }
+
+    ngAfterViewInit(): void {
+        const startDate = new Date();
+        startDate.setDate(startDate.getDate() - 5); // start from 5 days ago
+        this.loadRewardsHistory(startDate, new Date());
+    }
+
+    public removePsVault = (reward: RewardDto) => reward.vault.toLowerCase() != "ps";
```

Comment:
please, use for exclusion Utils.isAutoStakeVault 
sorry for unclear requirements when we discussed it

Reply:
No problem, consider it underway